src/libsyntax/parse/parser.rs

```diff
@@ -853,6 +868,18 @@ impl<'a> Parser<'a> {
 
             let cm = self.sess.source_map();
             match (cm.lookup_line(self.span.lo()), cm.lookup_line(sp.lo())) {
+                (Ok(ref a), Ok(ref b)) if a.line != b.line && is_semi_suggestable => {
```

Comment:
Nit: the code positioning seems a bit strange - `is_semi_suggestable` and `expects_semi` are only needed here, but code calculating them is scattered across the function.

Reply:
`is_semi_suggestable` can be anywhere and I can move it closer to the `match`. When originally writing this, I thought that `expects_semi` _needed_ to happen before sorting and deduplication because I misread and thought it was also being mapped. I'll merge both into one variable.